modules/imgcodecs/src/loadsave.cpp

```diff
@@ -714,6 +715,19 @@ static bool imwrite_( const String& filename, const std::vector<Mat>& img_vec,
     bool code = false;
     try
     {
+        FILE* f = fopen( filename.c_str(), "wb" );
+        if (f == 0)
+        {
+            if (errno == EACCES)
+            {
+                CV_Error( Error::StsError, "permission denied" );
+            }
+        }
+        else
+        {
+            fclose(f);
+            remove(filename.c_str());
```

Comment:
There is some chance that we may get problem like this again: #12342
See general note.

Reply:
I believe this issue is not something caused by OpenCV itself but rather an inherent could-be problem that can occur in some OS's like Windows which don't allow mutual sharing of file. In the unlikely scenario that the program lands on this else condition, it will at worse create a temporary file that is not removed, imwrite will still not crash as remove will simply return non zero. I think this could occur if another process from OS accesses the temporary file in between fclose() and remove() function. I will try to generate this occurrence by building opencv without the condition imposed by if( !f ) and share my findings. Additionally, I could further add warning log if fclose() and remove() do not return 0.